### ISSUE TITLE
Potential incomplete postcss configuration on the Next.js guide

### ISSUE THREAD

**Issue by awareness481** [tailwindlabs/tailwindcss.com#660]
On the [tailwindcss guide for Next.js](https://tailwindcss.com/docs/guides/nextjs#create-your-configuration-files) the `postcss.config.js` is the following:

```js
// postcss.config.js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```

However, on the [next.js documentantion for postcss](https://nextjs.org/docs/advanced-features/customizing-postcss-config#customizing-plugins) it says that when you define a **custom** postcss config file, the default behavior is disabled.

> Warning: When you define a custom PostCSS configuration file, Next.js completely disables the default behavior.

Based on the Next.js documentantion shouldn't the postcss.config.js file in the Tailwind docs be something like:

```js
{
  "plugins": [
    "postcss-flexbugs-fixes",
    [
      "postcss-preset-env",
        "stage": 3,
        "features": {
          "custom-properties": false
        }
      }
    ]
  ]
}
```

to ensure that the Next.js default behavior is not disabled? (except autoprefixer).

I'm not sure what tailwind takes care by default so if possible, a clarification on this would be great.

**Comment by adamwathan** [comment]
Hey! You only need those plugins if you actually want them. We don't use either of those two plugins on any of our own projects for example. You're correct that Next includes them by default though 👍🏻 

postcss-preset-env unfortunately has some bugs that cause issues when using certain Tailwind features (like `focus-visible`) so we don't go out of our way to encourage using it. Since styling is HTML-driven with Tailwind, it's not really important that you have the extra CSS-based features available that postcss-preset-env adds. Hope that helps!

**Comment by awareness481** [comment]
Yes, that makes it quite clear. I was worried something might 'break' once going into production. Thank you! :1st_place_medal: 

FWIW, I was already going to close the issue when Next.js Tailwind built-in support lands :P

**Comment by nvh95** [comment]
Thank you guys for this issue, I came to the same problem when setting up Next.js and tailwind. I want to add some more detail if someone has a same question as me.
[Follow tailwindcss (with Next.js) guide](https://tailwindcss.com/docs/guides/nextjs), we will have this file
```
// postcss.config.js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```
[As mentioned in Next.js docs](https://nextjs.org/docs/advanced-features/customizing-postcss-config#customizing-plugins), when we have a custom PostCSS config file, we will disable the default behavior, they are:
```
1. Autoprefixer automatically adds vendor prefixes to CSS rules (back to IE11).
2. Cross-browser Flexbox bugs are corrected to behave like the spec.
3. New CSS features are automatically compiled for Internet Explorer 11 compatibility:
```
(the default config:
```js
{
  "plugins": [
    "postcss-flexbugs-fixes",
    [
      "postcss-preset-env",
      {
        "autoprefixer": {
          "flexbox": "no-2009"
        },
        "stage": 3,
        "features": {
          "custom-properties": false
        }
      }
    ]
  ]
}
```
)
It's obvious that we both have `autoprefixer` in both cases. 
For the number 2, next.js uses [postcss-flexbugs-fixes](https://github.com/luisrudge/postcss-flexbugs-fixes/blob/683560e1f0a4e67009331b564d530ccfefb831ad/index.js#L31-L39), it fixes bugs [4](https://github.com/philipwalton/flexbugs/blob/master/README.md#4-flex-shorthand-declarations-with-unitless-flex-basis-values-are-ignored), [6](https://github.com/philipwalton/flexbugs/blob/master/README.md#6-the-default-flex-value-has-changed), and [8.1a](https://github.com/philipwalton/flexbugs/blob/master/README.md#8-flex-basis-doesnt-support-calc). All of them affects IE only. So if you are not targeting IE, it's totally fine to ignore.
For the number 3, if you don't support IE, you can ignore.
To sum up, if your app doesn't support IE, configure as in [tailwindcss docs](https://tailwindcss.com/docs/guides/nextjs) is sufficient.
